spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/JdbcDeleteQueryCreator.java

```diff
@@ -139,19 +135,28 @@ private void deleteRelations(List<Delete> deleteChain, RelationalPersistentEntit
 
 			if (aggregatePath.isEntity() && !aggregatePath.isEmbedded()) {
 
-				SqlContext sqlContext = new SqlContext(aggregatePath.getLeafEntity());
+				SqlContext sqlContext = new SqlContext();
```

Comment:
This `if` above looks suspicious. What if the Id is embedded? Could that be an early return? 
I think that `isEntity` should only return `true` if it is an actual entity. Also, we could consider introducing `AggregatePath.isIdentifier()` returning `true` for singular and embedded identifiers. In any case, this requires some fine-tuning so that (un)educated readers can immediately identify what is going on without diving into deeply nested branches.